packages/admin-next/dashboard/src/routes/orders/returns-create/components/create-returns/returns-form.tsx

```diff
@@ -0,0 +1,410 @@
+      const unitRefundable =
+        (curr.refundable || 0) / (curr.quantity - curr.returned_quantity)
+
+      return acc + unitRefundable * quantity[curr.id]
+    }, 0)
+
+    const amount = itemTotal - (shippingPrice || 0)
+    onRefundableAmountChange(amount)
+
+    return amount
+  }, [items, quantity, shippingPrice])
+
+  useEffect(() => {
+    form.setValue("enable_custom_shipping_price", false)
+    form.setValue("custom_shipping_price", 0)
+  }, [form.watch("enable_custom_refund")])
+
+  return (
+    <div className="flex size-full flex-col items-center overflow-auto p-16">
+      <div className="flex w-full max-w-[736px] flex-col justify-center px-2 pb-2">
+        <div className="flex flex-col gap-y-1 pb-10">
+          <Heading className="text-2xl">{t("general.details")}</Heading>
+        </div>
+        <Heading className="mb-2 text-base">
+          {t("orders.returns.chooseItems")}
+        </Heading>
+        {items.map((item) => (
+          <ReturnItem
+            key={item.id}
+            item={item}
+            form={form}
+            currencyCode={order.currency_code}
+          />
+        ))}
+
+        <div className="flex flex-col gap-y-1 pb-4 pt-8">
+          <Heading className="text-base">{t("fields.shipping")}</Heading>
+        </div>
+
+        <div className="flex gap-x-4">
+          <div className="flex-1">
+            <Text weight="plus" className="txt-small">
+              {t("fields.location")}
+            </Text>
+            <Text className="text-ui-fg-subtle txt-small mb-1">
+              {t("orders.returns.locationDescription")}
+            </Text>
+            <Form.Field
+              control={form.control}
+              name="location"
+              render={({ field: { onChange, ref, ...field } }) => {
+                return (
+                  <Form.Item>
+                    <Form.Control>
+                      <Select onValueChange={onChange} {...field}>
+                        <Select.Trigger className="bg-ui-bg-base" ref={ref}>
+                          <Select.Value />
+                        </Select.Trigger>
+                        <Select.Content>
+                          {stock_locations.map((l) => (
+                            <Select.Item key={l.id} value={l.id}>
+                              {l.name}
+                            </Select.Item>
+                          ))}
+                        </Select.Content>
+                      </Select>
+                    </Form.Control>
+                    <Form.ErrorMessage />
+                  </Form.Item>
+                )
+              }}
+            />
+          </div>
+          <div className="flex-1">
+            <Text weight="plus" className="txt-small">
+              {t("fields.shipping")}
+            </Text>
+            <Text className="text-ui-fg-subtle txt-small mb-1">
+              {t("orders.returns.shippingDescription")}
+            </Text>
```

Comment:
move these too <Form.Label> and <Form.Hint> so they work correctly for screen readers